mipengine/controller/algorithm_executor.py

```diff
@@ -500,341 +591,303 @@ def run_udf_on_local_nodes(
-        tables_data = []
-        for node, table_name in self.nodes_tables.items():
-            tables_data.append(node.get_table_data(table_name))
-        tables_data_flat = [table_data.columns for table_data in tables_data]
-        tables_data_flat = [
-            elem
-            for table in tables_data_flat
-            for column in table
-            for elem in column.data
-        ]  # TODO bejesus..
-        return tables_data_flat
-
-    def __repr__(self):
-        r = "LocalNodeTable:\n"
-        r += f"schema: {self.get_table_schema()}\n"
-        for node, table_name in self.nodes_tables.items():
-            r += f"{node} - {table_name} \ndata(LIMIT 20):\n"
-            rows = list(zip(*node.get_table_data(table_name).columns))
-            tmp = [str(row) for row in rows[0:20]]
-            r += "\n".join(tmp)
-            r += "\n"
-        return r
-
-    def _validate_matching_table_names(self, table_names: List[_TableName]):
-        table_name_without_node_id = table_names[0].without_node_id()
-        for table_name in table_names:
-            if table_name.without_node_id() != table_name_without_node_id:
-                return False
-        return True
-
-    class MismatchingTableNamesException(Exception):
-        def __init__(self, table_names):
-            self.message = f"Mismatched table names ->{table_names}"
-
-
-class _GlobalNodeTable(_INodeTable):
-    def __init__(self, node_table: Dict["_Node", "_TableName"]):
-        self._node_table = node_table
-
-    @property
-    def node_table(self):
-        return self._node_table
-
-    # TODO this is redundant, either remove it or overload all node methods here?
-    def get_table_schema(self):
-        node = list(self.node_table.keys())[0]
-        table_name: _TableName = list(self.node_table.values())[0]
-        table_schema: TableSchema = node.get_table_schema(table_name).columns
-        return table_schema
-
-    def get_table_data(self):  # -> {Node:TableData}
-        node = list(self.node_table.keys())[0]
-        table_name: _TableName = list(self.node_table.values())[0]
-        table_data = [column.data for column in node.get_table_data(table_name).columns]
-        return table_data
-
-    def __repr__(self):
-        node = list(self.node_table.keys())[0]
-        table_name: _TableName = list(self.node_table.values())[0]
-        r = f"GlobalNodeTable: {table_name.full_table_name}"
-        r += f"\nschema: {self.get_table_schema()}"
-        r += f"\ndata (LIMIT 20): \n"
-        tmp = [str(row) for row in self.get_table_data()[0:20]]
-        r += "\n".join(tmp)
-        return r
+        elif isinstance(algoexec_arg, Literal):
+            udf_argument = UDFArgument(kind=algoexec_arg.kind, value=algoexec_arg.value)
+        else:
+            # TODO specific exception
+            raise Exception(f"{type(algoexec_arg)=} is not accepted...")
+        return udf_argument
+
+
+def check_same_schema_tables(
```

Comment:
It is not a method, it is a function, so if the exception is thrown from within you would have to check the traceback to figure what called it and caused the exception, so it would make more sense to return some info about the tables queried about their schemas and based on that raise an exception if needed. It could serve as a small tool to check if a group of tables have the same schema and return the respective info about the tables' schemas. I anyway turned it to a method `_validate_same_schema_tables` and it now raises an exception if the schemas do not match